doc/api/fs.md

```diff
@@ -631,11 +627,34 @@ On Linux, positional writes do not work when the file is opened in append mode.
 The kernel ignores the position argument and always appends the data to
 the end of the file.
 
+#### `filehandle.write(buffer, options)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `buffer` {Buffer|TypedArray|DataView}
+* `options` {Object}
+  * `offset` {integer} **Default:** `0`
+  * `length` {integer} **Default:** `buffer.byteLength - offset`
+  * `position` {integer} **Default:** `null`
+* Returns: {Promise}
+
+Write `buffer` to the file.
+
+Similar to the above `filehandle.write` function, this version takes an
+optional `options` object. If no `options` object is specified, it will
+default with the above values.
+
 #### `filehandle.write(string[, position[, encoding]])`
 
 <!-- YAML
 added: v10.0.0
 changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/41677
+    description: The `string` parameter no longer accepts objects with an
+                 own `toString` function property.
   - version: v14.12.0
     pr-url: https://github.com/nodejs/node/pull/34993
     description: The `string` parameter will stringify an object with an
```

Comment:
In the end, did it ever accepted such objects? `node -e 'node -e 'fs.promises.open("/dev/null").then(fh => fh.write({ toString() { return "string" } }))'` results in an abort on all release lines (tested on v14.x, v16.x, and v17.x). If this was never supported, maybe we can simply fix the doc and remove the added entry (I realize you added this because I asked for it, sorry for the confusion, this `fs.md` file is too big, I got confused).
In the end, did it ever accepted such objects? `node -e 'node -e 'fs.promises.open("/dev/null").then(fh => fh.write({ toString() { return "string" } }))'` results in an abort on all release lines (tested on v14.x, v16.x, and v17.x). If this was never supported, maybe we can simply fix the doc and remove the added entry (I realize you added this because I asked for it, sorry for the confusion, this `fs.md` file is too big, I got confused).

```suggestion
```

Reply:
Yep, all affected methods weren't working with coercion in the first place. Annihilated all related entries.
I initially assumed that "rewriting history" in the docs would be a bad idea; but yes, it is much cleaner this way, and it won't mislead people who use older versions of Node.

> this `fs.md` file is too big

It sure is. 😄 I hope, something like removal of whole Callback API in foreseeable future will reduce it, and both code and docs will become less confusing.